modules/caddyhttp/replacer.go

```diff
@@ -335,16 +336,32 @@ func getReqTLSReplacement(req *http.Request, key string) (any, bool) {
 			switch {
 			case strings.HasPrefix(field, "dns_names"):
 				fieldName = "dns_names"
-				fieldValue = cert.DNSNames
+				if isTLSReq {
+					fieldValue = cert.DNSNames
+				} else {
+					return nil, true
+				}
```

Comment:
In general, return as early as possible. Don't use `else` if you can avoid it. This applies to many places in this PR, please apply this idea throughout.

```suggestion
				if !isTLSReq {
					return nil, true
				}
				fieldName = "dns_names"
				fieldValue = cert.DNSNames
```

Reply:
fixed all same parts